SurveyAppTests/Tests/Services/Network/API/BaseAPISpec.swift

```diff
@@ -0,0 +1,121 @@
+        
+        describe("Test BaseAPI perform request") {
+            beforeEach {
+                userSession = UserSessionProtocolMock()
+                api = BaseAPI(userSession: userSession)
+            }
+            
+            afterEach {
+                HTTPStubs.removeAllStubs()
+            }
+            
+            context("when making a valid request") {
+                beforeEach {
+                    stub(condition: isMethodPOST()) { _ in
+                        HTTPStubsResponse(data: sampleSuccessData, statusCode: 200, headers: [:])
+                    }
+
+                    configuration = RequestConfiguration(
+                        method: .post,
+                        url: "\(Defines.baseURL)/test"
+                    )
+                }
+                
+                it("returns a decoded object") {
+                    waitUntil(timeout: .seconds(2)) { done in
+                        _ = api.performRequest(with: configuration) { (result: Result<UserCredential, APIError>) in
+                            switch result {
+                            case .success(let credentials):
+                                expect(credentials.accessToken) == "sampleAccessToken"
+                                expect(credentials.refreshToken) == "sampleRefreshToken"
+                            case .failure:
+                                break
+                            }
+                            done()
+                        }
+                    }
+                }
+            }
+            
+            context("when making an invalid request") {
+                beforeEach {
+                    stub(condition: isMethodPOST()) { _ in
+                        HTTPStubsResponse(data: sampleFailureData, statusCode: 400, headers: [:])
+                    }
+
+                    configuration = RequestConfiguration(
+                        method: .post,
+                        url: "\(Defines.baseURL)/test"
+                    )
+                }
+                
+                it("returns an APIError object") {
+                    waitUntil(timeout: .seconds(2)) { done in
+                        _ = api.performRequest(with: configuration) { (result: Result<UserCredential, APIError>) in
+                            switch result {
+                            case .success:
+                                break
+                            case .failure(let error):
+                                expect(error).to(beAKindOf(APIError.self))
+                                expect(error.errors?[0].detail) == "Sample API error detail"
+                            }
+                            done()
+                        }
+                    }
+                }
+            }
+        }
+        
+        describe("Test BaseAPI url") {
+            context("when passing a URL path") {
+                it("creates a valid url") {
```

Comment:
The tip here that when you combine `describe, context, it` they should make a sentence
So here is your sentence now
```
Test BaseAPI URL when passing a URL path, it creates a valid URL
```
So I think we all want to say
```
A BaseAPI when passing a URL path, it creates a valid URL
```

Reply:
Then you may want to update for the rest 😬